### ISSUE TITLE
Write to HTTPOnly cookie from server.R

### ISSUE THREAD

**Issue by saikatmitra71** [rstudio/shiny#3524]
I want to add/modify HTTPOnly cookies from server.R. I've tried session$request$HTTP_COOKIE <- paste0(session$request$HTTP_COOKIE[1], "; t=hello") where "t=hello" is my desired cookie-value pair, but it doesn't work. It prints as result, but doesn't actually add to cookies which can be checked from browser.

```
library(shiny)

ui <- fluidPage(
  textOutput("cookie"),
  br(),
  tableOutput("table")
)

server <- function(input, output, session){
  
  session$request$HTTP_COOKIE <- paste0(session$request$HTTP_COOKIE[1], "; t=hello")
  
  output$cookie <- renderPrint({
    session$request$HTTP_COOKIE
  })
  
}

shinyApp(ui, server)
```

Any kind of guideline would be very much helpful.

**Comment by jcheng5** [comment]
It's not impossible, but it does take some surprising gymnastics; the communication between the web page and server code is mostly through a websocket, so there's not a natural place to attach a cookie header. Can you say more about what you're trying to achieve with this cookie?

**Comment by karawoo** [comment]
Hope you don't mind me chiming in on this old issue. Since the OP didn't respond with their use case, I have one from a situation where we'd like to be able to delete or modify an HttpOnly cookie from shiny. In our case we've built authentication into our app using a separate authentication service on the same domain as our shiny app. When a user visits the shiny app they're immediately redirected to that service's login page, they enter their credentials and are redirected back to our shiny app, now with a JWT access token set in an HttpOnly cookie. We'd like to be able to delete (or expire) that cookie when the user logs out, but haven't figured out how to do so yet. I'm curious what the gymnastics to do this would be, or if it's just not possible.

**Comment by jcheng5** [comment]
Hi Kara, can you tell me what “the user logs out” means? A link/button in the Shiny app?

**Comment by karawoo** [comment]
Yeah exactly, we'd like to add a logout button in the app that would delete or expire the cookie and redirect the user to the login page.

**Comment by jcheng5** [comment]
It's doable but my first choice would be to link to a logout URL/API that's hosted by your authentication service, is that possible? It seems weird that the auth service would contain the logic for logging in, but your application would contain the logic for logging out; even just from a logging/audit perspective, I'd imagine you want the auth service to handle both?

**Comment by karawoo** [comment]
Yeah, there is a logout API that I'd call as well, but for whatever reason it doesn't remove the cookies (even though the login API sets them). Maybe I can just make that feature request instead of trying to work it out in the shiny app. Thanks for the feedback!

**Comment by jcheng5** [comment]
I would for sure try to get that to work first.

If you really cannot go that way, then some clues as to how to do this in Shiny are here: https://gist.github.com/jcheng5/2aaff19e67079840350d08361fe7fb20 (Don't miss the "Handling URL paths besides `/`" section at the end, I suspect you'll want to do this with a GET request to a custom path.)
